planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h

```diff
@@ -136,7 +137,11 @@ class CurrentStateMonitor
    *  @return Returns the map from joint names to joint state values*/
   std::map<std::string, double> getCurrentStateValues() const;
 
-  /** @brief Wait for at most \e wait_time seconds until the complete current state is known. Return true if the full state is known */
+  /** @brief Wait for at most \e wait_time seconds for a robot state more recent than t */
+  bool waitForCurrentState(const ros::Time t=ros::Time::now(), double wait_time=1) const;
+
+  // TODO: rename functions to waitForCompleteState
+  /** @brief Wait for at most \e wait_time seconds until the complete robot state is known. Return true if the full state is known */
   bool waitForCurrentState(double wait_time) const;
```

Comment:
We *could* mark this function as deprecated in indigo, to point out that it does not what it says.
We can't remove the function in indigo, but I think it is a good idea to mark it as deprecated and
make it call the renamed function `waitForCompleteState` in indigo.